soundfile.py

```diff
@@ -614,6 +614,8 @@ def __init__(self, file, mode='r', samplerate=None, channels=None,
         >>> assert myfile.closed
 
         """
+        # resolve pathlib.Path objects:
```

Comment:
As we've discussed, this isn't (only) about `Path` objects.
You should probably mention `PathLike` or simply "PEP519".

You could also mention that, once Python < 3.6 isn't supported anymore (in the far future), this line can be replaced with:

    file = _os.fspath(file)